### PR TITLE
WINC-1310: Remove certain tests in old upgrade suite

### PR DESCRIPTION
**Pull request by sebsoto**
Removes the old upgrade test, as it has been replaced by the newer upgrade test for a while now. The reconfigure test was moved to run on other cloud platforms as that was the only part of the test run that is not covered elsewhere.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -242,8 +242,7 @@ run-ci-e2e-byoh-test:
 	hack/run-ci-e2e-test.sh -t basic -m 0
 
 .PHONY: run-ci-e2e-upgrade-test
-run-ci-e2e-upgrade-test:
-	hack/run-ci-e2e-test.sh -t upgrade
+run-ci-e2e-upgrade-test: ;
```
Comment (jrvaldes) [inline Makefile:245]: Consider removing the corresponding job in the release repo https://github.com/openshift/release/blob/master/ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-release-4.17.yaml#L248

**Review by jrvaldes — APPROVED** [review]
/lgtm

File: hack/run-ci-e2e-test.sh [hunk 2]
```diff
@@ -163,16 +163,8 @@ if [[ "$TEST" = "all" || "$TEST" = "basic" ]]; then
   go test ./test/e2e/... -run=TestWMCO/service_reconciliation -v -timeout=20m -args $GO_TEST_ARGS
   printf "\n####### Testing cluster-wide proxy #######\n" >> "$ARTIFACT_DIR"/wmco.log
   go test ./test/e2e/... -run=TestWMCO/cluster-wide_proxy -v -timeout=20m -args $GO_TEST_ARGS
-fi
-
-if [[ "$TEST" = "all" || "$TEST" = "upgrade" ]]; then
```
[the remark this answers is not in the document]
Comment (sebsoto) [reply hack/run-ci-e2e-test.sh:168]: That was already replaced with a stub to allow the test to pass for merge, but I removed all other uses of `upgrade` in this file.

**Pull request merged**